tests/test_settings.py

```diff
@@ -11,21 +11,41 @@
 
 
 class TestThing(lt.Thing):
-    boolsetting = lt.ThingSetting(bool, False, description="A boolean setting")
-    stringsetting = lt.ThingSetting(str, "foo", description="A string setting")
-    dictsetting = lt.ThingSetting(
-        dict, {"a": 1, "b": 2}, description="A dictionary setting"
-    )
+    """A test `.Thing` with some settings and actions."""
 
-    _float = 1.0
+    def __init__(self) -> None:
+        super().__init__()
+        # Initialize functional settings with default values
+        self._floatsetting: float = 1.0
 
-    @lt.thing_setting
+    boolsetting: bool = lt.setting(default=False)
+    "A boolean setting"
+
+    stringsetting: str = lt.setting(default="foo")
+    "A string setting"
+
+    dictsetting: dict = lt.setting(default_factory=lambda: {"a": 1, "b": 2})
+    "A dictionary setting"
+
+    @lt.setting
     def floatsetting(self) -> float:
-        return self._float
+        """A float setting."""
+        return self._floatsetting
 
     @floatsetting.setter
     def floatsetting(self, value: float):
-        self._float = value
+        self._floatsetting = value
+
+    @lt.setting
+    def readonlysetting(self) -> str:
+        """A read-only setting."""
+        return "This is read-only"
+
+    @readonlysetting.setter
+    def readonlysetting(self, value: str):
+        pass
```

Comment:
If we are testing readonly but with a setter it would be good to make it functional so that we can check it with more in-depth tests. 
"This is read-only"
```suggestion

    .self._readonlysetting: str = "This is read-only"

    @lt.setting
    def readonlysetting(self) -> str:
        """A read-only setting."""
        return self._readonlysetting

    @readonlysetting.setter
    def readonlysetting(self, value: str):
        self._readonlysetting = value
```

Reply:
Happy to make it functional. Is `.self.` a language feature I've missed? A quick google (and typing it into my python interpreter) doesn't leave me any the wiser...